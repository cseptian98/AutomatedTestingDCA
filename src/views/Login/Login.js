import React, {useEffect, useState} from 'react'
import {
  Text,
  View,
  Image,
  TextInput,
  TouchableOpacity,
  StatusBar,
  ActivityIndicator,
  Alert,
} from 'react-native'
import styles from './Login.styles'
import Icon from 'react-native-vector-icons/MaterialCommunityIcons'
import {storeData, getData} from 'api/Local'
import axiosConfig, {setToken} from 'api/BaseConfig'
import {StackActions} from '@react-navigation/native'
import {
  TEST_ID_IMAGE_LOGIN,
  TEST_ID_EMAIL_LOGIN,
  TEST_ID_PASSWORD_LOGIN,
  TEST_ID_BUTTON_LOGIN,
  TEST_ID_BUTTON_CREATE_ACCOUNT,
} from 'constants'
import {testProps} from 'utils/testProps.helper'

const LoginScreen = ({navigation}) => {
  const [email, setEmail] = useState('')
  const [value, setValue] = useState(null)
  const [password, setPassword] = useState('')
  const [isLoading, setIsLoading] = useState(false)
  const [showPassword, setShowPassword] = useState(true)

  const readFromStorage = async () => {
    const data = await getData()
    setValue(data)
  }

  useEffect(() => {
    readFromStorage()
  }, [])

  if (value) {
    setToken(value.token)
    navigation.replace('Home')
  }

  const doLogin = () => {
    setIsLoading(true)

    const onSuccess = ({data}) => {
      setIsLoading(false)
      storeData(data)
      setToken(data.token)
      navigation.replace('Home')
    }

    const onFailure = () => {
      Alert.alert('Error', 'Login Error', [
        {
          text: 'Close',
          onPress: () => setIsLoading(false),
        },
      ])
    }

    axiosConfig
      .post('api/Auth/login', {email, password})
      .then(onSuccess)
      .catch(onFailure)
  }

  const toRegister = () => {
    const movePage = StackActions.push('Register', {url: 'api/Auth/register'})
    navigation.dispatch(movePage)
  }

  const setIconPassword = () => {
    setShowPassword(!showPassword)
  }

  return (
    <View style={styles.container}>
      <Image
        source={require('assets/images/login.png')}
        style={styles.image}
        {...testProps(TEST_ID_IMAGE_LOGIN)}
      />

      <StatusBar style="auto" />
      <View style={styles.inputView}>
        <TextInput
          keyboardType="email-address"
          onChangeText={value => setEmail(value)}
          placeholder="Email"
          placeholderTextColor="#000"
          style={styles.textInput}
          value={email}
          {...testProps(TEST_ID_EMAIL_LOGIN)}
        />
      </View>

      <View style={styles.inputView}>
        <TextInput
          onChangeText={value => setPassword(value)}
          placeholder="Password"
          placeholderTextColor="#000"
          secureTextEntry={showPassword}
          style={styles.textInput}
          value={password}
          {...testProps(TEST_ID_PASSWORD_LOGIN)}
        />
        <TouchableOpacity onPress={setIconPassword} style={styles.icon}>
          {showPassword ? (
            <Icon name="eye-off" size={24} color="#000" />
          ) : (
            <Icon name="eye" size={24} color="#000" />
          )}
        </TouchableOpacity>
      </View>
      {isLoading ? (
        <ActivityIndicator size="large" color="#0D47A1" />
      ) : (
        <>
          <TouchableOpacity
            onPress={doLogin}
            style={styles.loginButton}
            {...testProps(TEST_ID_BUTTON_LOGIN)}>
            <Text style={styles.textButton}>Login</Text>
          </TouchableOpacity>
          <TouchableOpacity
            onPress={toRegister}
            style={styles.registerButton}
            {...testProps(TEST_ID_BUTTON_CREATE_ACCOUNT)}>
            <Text style={styles.textButton2}>Create Account</Text>
          </TouchableOpacity>
        </>
      )}
    </View>
  )
}

export default LoginScreen
