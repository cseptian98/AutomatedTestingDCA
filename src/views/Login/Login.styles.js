import {StyleSheet} from 'react-native';

const styles = StyleSheet.create({
  container: {
    flex: 1,
    alignItems: 'center',
    justifyContent: 'flex-start',
    backgroundColor: '#FFF',
  },
  image: {
    width: 280,
    height: 180,
    marginBottom: 30,
    marginTop: 40,
  },
  inputView: {
    height: 45,
    width: '80%',
    marginBottom: 20,
    borderBottomWidth: 1,
    borderBottomColor: '#000',
  },
  textInput: {
    flex: 1,
    height: 30,
  },
  textButton: {
    color: '#212121',
  },
  forgotButton: {
    height: 30,
    marginBottom: 30,
  },
  loginButton: {
    height: 50,
    width: '80%',
    marginTop: 20,
    color: '#002F6C',
    borderRadius: 25,
    alignItems: 'center',
    justifyContent: 'center',
    backgroundColor: '#FFD500',
  },
  registerButton: {
    height: 50,
    width: '80%',
    marginTop: 20,
    borderRadius: 25,
    alignItems: 'center',
    justifyContent: 'center',
    backgroundColor: '#eceff1',
  },
});

export default styles;
