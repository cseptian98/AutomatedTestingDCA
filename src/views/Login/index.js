import React, {useState, useEffect} from 'react'
import {
  Text,
  View,
  Image,
  TextInput,
  TouchableOpacity,
  StatusBar,
  ActivityIndicator,
  Alert,
} from 'react-native'
import styles from './Login.styles'
import {storeData, getData} from 'api/Local'
import axiosConfig, {setToken} from 'api/BaseConfig'
import {StackActions} from '@react-navigation/native'

const Login = ({navigation}) => {
  const [email, setEmail] = useState('')
  const [value, setValue] = useState(null)
  const [password, setPassword] = useState('')
  const [isLoading, setIsLoading] = useState(false)

  const readFromStorage = async () => {
    const data = await getData()
    setValue(data)
  }

  useEffect(() => {
    readFromStorage()
  }, [])

  if(value) {
    setToken(value.token)
    navigation.replace('Home')
  }

  const doLogin = () => {
    setIsLoading(true)
    const onSuccess = ({data}) => {
      setIsLoading(false)
      storeData(data)
      setToken(data.token)
      navigation.replace('Home')
    }

    const onFailure = data => {
      console.log('debug error', data)
      Alert.alert('Login Error', 'Error Message', [
        {
          text: 'Close',
          onPress: () => setIsLoading(false),
        },
      ])
    }

    axiosConfig
      .post('api/Auth/login', {email, password})
      .then(onSuccess)
      .catch(onFailure)
  }

  const toRegister = () => {
    const movePage = StackActions.push('Register')
    navigation.dispatch(movePage)
  }

  return (
    <View style={styles.container}>
      <Image style={styles.image} source={require('assets/images/login.png')} />

      <StatusBar style="auto" />
      <View style={styles.inputView}>
        <TextInput
          style={styles.textInput}
          placeholder="Email"
          placeholderTextColor="#000"
          keyboardType="email-address"
          onChangeText={email => setEmail(email)}
        />
      </View>

      <View style={styles.inputView}>
        <TextInput
          style={styles.textInput}
          placeholder="Password"
          placeholderTextColor="#000"
          secureTextEntry
          onChangeText={password => setPassword(password)}
        />
      </View>
      {isLoading ? (
        <ActivityIndicator size="large" color="#0D47A1" />
      ) : (
        <>
          <TouchableOpacity style={styles.loginButton} onPress={doLogin}>
            <Text style={styles.textButton}>Login</Text>
          </TouchableOpacity>
          <TouchableOpacity style={styles.registerButton} onPress={toRegister}>
            <Text style={styles.textButton2}>Create Account</Text>
          </TouchableOpacity>
        </>
      )}
    </View>
  )
}

export default Login
