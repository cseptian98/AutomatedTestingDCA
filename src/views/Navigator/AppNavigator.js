import React from 'react'
import {createStackNavigator} from '@react-navigation/stack'
import Login from 'views/Login'
import Home from 'views/Home'
import Register from 'views/Register'

const Stack = createStackNavigator()

const AppNavigator = () => {
  return (
    <Stack.Navigator initialRouteName="Login">
      <Stack.Screen name="Login" component={Login} />
      <Stack.Screen
        name="Home"
        component={Home}
        options={{title: 'Automated Testing'}}
      />
      <Stack.Screen name="Register" component={Register} />
    </Stack.Navigator>
  )
}

export default AppNavigator
