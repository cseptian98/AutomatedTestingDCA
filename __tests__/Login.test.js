import React from 'react'
import {fireEvent, render, waitFor} from '@testing-library/react-native'
import {NavigationContainer} from '@react-navigation/native'
import AppNavigator from 'views/Navigator/AppNavigator'
import Login from 'views/Login'

describe('Login', () => {
  it('test execution', async () => {
    const component = (
      <NavigationContainer>
        <AppNavigator />
      </NavigationContainer>
    );
    const root = await waitFor(() => {
      render(<Login />)
    })
    const {getByTestId} = root
    fireEvent.changeText(getByTestId('email'), 'cseptian@gmail.com')
    fireEvent.changeText(getByTestId('password'), 'Auto@123')

    // fireEvent.press(getByText('Login'))
    // expect(onPressMock).toHaveBeenCalled()

    // expect(getByTestId('email')).stringContaining('cseptian@gmail.com')
    // expect(getByTestId('password')).stringContaining('Auto@123')
  })
})
