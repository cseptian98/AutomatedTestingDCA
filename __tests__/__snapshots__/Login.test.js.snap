// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Login Components renders correctly 1`] = `
<View
  style={
    Object {
      "alignItems": "center",
      "backgroundColor": "#FFF",
      "flex": 1,
      "justifyContent": "flex-start",
    }
  }
>
  <Image
    accessibilityLabel="TEST_ID_IMAGE_LOGIN"
    source={
      Object {
        "testUri": "../../../src/assets/images/login.png",
      }
    }
    style={
      Object {
        "height": 200,
        "marginBottom": 16,
        "marginTop": 24,
        "width": 280,
      }
    }
    testID="TEST_ID_IMAGE_LOGIN"
  />
  <View
    style={
      Object {
        "borderBottomColor": "#212121",
        "borderBottomWidth": 1,
        "flexDirection": "row",
        "margin": 10,
        "width": "90%",
      }
    }
  >
    <TextInput
      accessibilityLabel="TEST_ID_EMAIL_LOGIN"
      allowFontScaling={true}
      keyboardType="email-address"
      onChangeText={[Function]}
      placeholder="Email"
      placeholderTextColor="#000"
      rejectResponderTermination={true}
      style={
        Object {
          "flex": 1,
        }
      }
      testID="TEST_ID_EMAIL_LOGIN"
      underlineColorAndroid="transparent"
      value=""
    />
  </View>
  <View
    style={
      Object {
        "borderBottomColor": "#212121",
        "borderBottomWidth": 1,
        "flexDirection": "row",
        "margin": 10,
        "width": "90%",
      }
    }
  >
    <TextInput
      accessibilityLabel="TEST_ID_PASSWORD_LOGIN"
      allowFontScaling={true}
      onChangeText={[Function]}
      placeholder="Password"
      placeholderTextColor="#000"
      rejectResponderTermination={true}
      secureTextEntry={true}
      style={
        Object {
          "flex": 1,
        }
      }
      testID="TEST_ID_PASSWORD_LOGIN"
      underlineColorAndroid="transparent"
      value=""
    />
    <View
      accessible={true}
      focusable={true}
      onClick={[Function]}
      onResponderGrant={[Function]}
      onResponderMove={[Function]}
      onResponderRelease={[Function]}
      onResponderTerminate={[Function]}
      onResponderTerminationRequest={[Function]}
      onStartShouldSetResponder={[Function]}
      style={
        Object {
          "alignItems": "center",
          "justifyContent": "center",
          "opacity": 1,
        }
      }
    >
      <Text
        allowFontScaling={false}
        style={
          Array [
            Object {
              "color": "#000",
              "fontSize": 24,
            },
            undefined,
            Object {
              "fontFamily": "Material Design Icons",
              "fontStyle": "normal",
              "fontWeight": "normal",
            },
            Object {},
          ]
        }
      >
        ó°ˆ‰
      </Text>
    </View>
  </View>
  <View
    accessibilityLabel="TEST_ID_BUTTON_LOGIN"
    accessible={true}
    focusable={true}
    onClick={[Function]}
    onResponderGrant={[Function]}
    onResponderMove={[Function]}
    onResponderRelease={[Function]}
    onResponderTerminate={[Function]}
    onResponderTerminationRequest={[Function]}
    onStartShouldSetResponder={[Function]}
    style={
      Object {
        "alignItems": "center",
        "backgroundColor": "#0D47A1",
        "borderRadius": 16,
        "height": 50,
        "justifyContent": "center",
        "marginTop": 20,
        "opacity": 1,
        "width": "90%",
      }
    }
    testID="TEST_ID_BUTTON_LOGIN"
  >
    <Text
      style={
        Object {
          "color": "#FFF",
          "fontSize": 16,
        }
      }
    >
      Login
    </Text>
  </View>
  <View
    accessibilityLabel="TEST_ID_CREATE_ACCOUNT"
    accessible={true}
    focusable={true}
    onClick={[Function]}
    onResponderGrant={[Function]}
    onResponderMove={[Function]}
    onResponderRelease={[Function]}
    onResponderTerminate={[Function]}
    onResponderTerminationRequest={[Function]}
    onStartShouldSetResponder={[Function]}
    style={
      Object {
        "alignItems": "center",
        "backgroundColor": "#E3F2FD",
        "borderRadius": 16,
        "height": 50,
        "justifyContent": "center",
        "marginTop": 20,
        "opacity": 1,
        "width": "90%",
      }
    }
    testID="TEST_ID_CREATE_ACCOUNT"
  >
    <Text
      style={
        Object {
          "color": "#212121",
          "fontSize": 16,
        }
      }
    >
      Create Account
    </Text>
  </View>
</View>
`;
